datasets.py

```diff
@@ -117,6 +122,89 @@ def read_xml_content(xml_file):
     11: 'indoor'
 }
 
+def group_mapping_creator(labels_to_names, supercategories_to_names, 
+                          override_map = None):
+    '''
+    inputs:
+    labels_to_names: dict mapping "label" to "human readable string"
+    supercategories_to_names: dict mapping "supercat" to "human readable string"
+    override_map: dict mapping: "human-readable label" to "human-readable supercat"
+                        for manual overriding of mapping of certain labels
+           
+    output:
+    prints out human-readable label to human readable supercat mapping
+    
+    returns:
+    function that takes an input label, and returns a supercat
+    '''
+    ######################################################################  
+    assert (labels_to_names is not None and supercategories_to_names is not None)
+    # function that calculates the "closest" supercategory to input word
+    # and returns said supercategory and the associated softmax confidence score
+    def dist_calculator(word):
+        assert(word is not None)
+        # create list of supercats for easy indexing & ordering
+        supercat_list = list(supercategories_to_names.keys())
+        # score array for each supercat, for softmax calculation
+        score_arr = []
+
+        for supercat in (supercat_list):
+            word1 = nlp(supercategories_to_names[supercat])
+            word2 = nlp(word)
+            cur_score = word1.similarity(word2)
+            if cur_score is None:
+                score_arr.append(-1)
+                continue
+            else:
+                score_arr.append(cur_score)
+        score_arr = softmax(score_arr)
+        
+        return supercat_list[np.argmax(score_arr)], np.max(score_arr)
+    ######################################################################    
+    # this represents the result map from label to supercat
+    result_label_to_group_map = OrderedDict()
+    # (hr_label) is the human-readable value corresponding to the key (label)
+    for label, hr_label in labels_to_names.items():
+        if override_map and hr_label in override_map:
+            # skip if user is overriding with override_map
+            continue
+        # retrieve tuple result from dist_calculator
+        supercat_match = dist_calculator(hr_label)
+        result_label_to_group_map[label] = supercat_match
+        
+    # sort from least to most confident based on softmax scores
+    result_label_to_group_map = dict(sorted(result_label_to_group_map.items(), key=lambda item: item[1][1]))
```

Comment:
I think you mean OrderedDict here and not dict?

Reply:
changed, thanks!